Remotion/Data/DomainObjects.UnitTests/Queries/Configuration/QueryConfigurationTest.cs

```diff
@@ -126,98 +147,75 @@ public void GetDefaultQueryFilePath_BaseDirectory ()
 #endif
     public void GetDefaultQueryFilePath_WithRelativeSearchPath ()
     {
-      AppDomainSetup setup = AppDomain.CurrentDomain.SetupInformation;
-      setup.ApplicationBase = Path.GetPathRoot (AppContext.BaseDirectory);
-      setup.DynamicBase = Path.GetTempPath ();
-      setup.PrivateBinPath = Path.GetFullPath (AppContext.BaseDirectory).Substring (setup.ApplicationBase.Length); // make a relative path
+      IAppContextProvider providerStub = MockRepository.GenerateStub<IAppContextProvider>();
+      providerStub.Stub (_ => _.BaseDirectory).Return (Path.GetPathRoot (_tempQueriesDirectory));
+      providerStub.Stub (_ => _.RelativeSearchPath).Return (Path.GetFullPath (_tempQueriesDirectory).Substring (providerStub.BaseDirectory.Length)); // make relative path
```

Comment:
We should not use stubbed values when setting up other stubs. Yes, sometimes it's convenient, and yes, there's code with this pattern in the code base, but it's still not a great practice. 

Reasoning: it can theoretically result in some weird bugs if there's an ordering issue in the mock setup. Plus, it may imply a logic duplication with the production code, which should also be avoided in tests.

Reply:
Changed in c6ec9d3cb9825d5cee347bbe71b1eb6bed142966.